src/devices/HCSR501/HCSR501.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Devices.Gpio;
+
+namespace Iot.Device.HCSR501
+{
+    public class HCSR501ValueChangedEventArgs : EventArgs
+    {
+        public readonly PinValue PinValue;
+        public HCSR501ValueChangedEventArgs(PinValue value)
+        {
+            PinValue = value;
+        }
+    }
+
+    public class HCSR501 : IDisposable
+    {
+        private GpioPin sensor;
```

Comment:
I believe you are still using the old API, in the new one, we don't have a pin object anymore, but instead we have just the integer representing the pin number.